### PR TITLE
[Feature] Added unit test for `NoviAction*` classes to test `Flow` comparisons

### LINKED ISSUE #2: Add a unit test covering Experimenter Actions making sure it won't impact on equality comparisons
This is to make sure that flows with Experimenter Actions should work seamlessly and don't impact in the [`__eq__` method](https://github.com/kytos-ng/of_core/blob/master/flow.py#L210-L216) that's used by `flow_manager`, since `flow_manager` will compare flows with Experimenter Actions we want to cover this case. 

Related to https://github.com/kytos-ng/of_core/issues/30

### PR DESCRIPTION
**Pull request by viniarck**
Fixes #2 

Originally, this was reported as an issue (requirement) on `of_core` repo https://github.com/kytos-ng/of_core/issues/30, but the unit tests related to Experimenter are on this repo, so took the opportunity to bootstrap the unit tests for this repo as well. 

Based on the tests added, the `__eq__`  method still works as expected so the flow comparions that `flow_manager` uses should also work as a result, so this mitigate the concerns that the team has raised regarding whether this would impact or not. 

Heads up:

- I had to also open this PR on `of_core` to have the unit tests passing, so this dependency needs to land first https://github.com/kytos-ng/of_core/pull/36

### CODE REVIEW

File: requirements/dev.txt [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+# This file is autogenerated by pip-compile with python 3.6
+# To update, run:
+#
+#    pip-compile --output-file=requirements/dev.txt requirements/dev.in
+#
+-e git+https://github.com/kytos-ng/kytos.git#egg=kytos
+    # via -r requirements/dev.in
+-e .
+    # via -r requirements/dev.in
+-e git+https://github.com/kytos-ng/of_core.git@fix/action_base_from_dict#egg=kytos_of_core
```
Comment (viniarck) [inline requirements/dev.txt:11]: This is a temporary commit, I'll revert this commit https://github.com/kytos-ng/noviflow/pull/3/commits/6bf478c2ae9e5cc8bc644696e1f1e9daabbfaf1b pointing back to `of_core` master branch before it gets approved, so it depends on getting merged https://github.com/kytos-ng/of_core/pull/36

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,27 @@
+[pycodestyle]
+max-line-length = 88
+exclude = .eggs,ENV,build,docs/conf.py,venv
+add-ignore = D105
+# D105: Missing docstring in magic method
+
+[yala]
+radon mi args = --min C
+pylint args = --disable==too-many-arguments,too-many-locals,too-few-public-methods,too-many-instance-attributes,no-else-return,dangerous-default-value,duplicate-code,raise-missing-from,too-many-arguments,too-many-public-methods
+linters=isort,pycodestyle,pylint # TODO enable black later to minimize changes
```
Comment (viniarck) [inline setup.cfg:10]: Added `black` but I haven't formatted the rest of the code with it, I'll do it in an upcoming PR to minimize changes and facilitating the review.

File: tests/unit/test_main.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+        from napps.amlight.noviflow.main import Main
+
+        self.addCleanup(patch.stopall)
+
+        controller = get_controller_mock()
+        self.napp = Main(controller)
+        self.napp.register_actions()
+        self.mock_switch = get_switch_mock("00:00:00:00:00:00:00:01", 0x04)
+
+    def test_create_noviactions(self):
```
Comment (viniarck) [inline tests/unit/test_main.py:34]: I'm planning to increase test cov to also add tests for unpacking on this issue here https://github.com/kytos-ng/noviflow/issues/4

**Review by ajoaoff — APPROVED** [review]

File: requirements/dev.txt (hunk 1 above)
Comment (viniarck) [reply requirements/dev.txt]: @ajoaoff I've just reverted this commit on https://github.com/kytos-ng/noviflow/pull/3/commits/feecc490744dd5d9ecfd8b4fe79f0cf4fe1de77c, and I've also applied some `pylint` fixes https://github.com/kytos-ng/noviflow/pull/3/commits/7049c1ae1f3f74466ab57115cc2d6a0df9ce7d31, could you do a second pass review? thanks

**Comment by viniarck** [comment]
thanks for reviewing and your help, Antonio.

**Pull request merged**